alf/config_util.py

```diff
@@ -448,6 +471,15 @@ def __init__(self, param1, param2='a default value'):
     In this case, the name of the configurable is 'MyClass', and both `param1`
     and `param2` are configurable.
 
+    The full name of a configurable value is MODULE_PATH.FUNC_NAME.ARG_NAME. It
+    can be referred using any suffixes as long as there is no ambiguity. For
+    example, assuming there are two configurable values "abc.def.func.a" and
+    "xyz.uvw.func.a", you can use "abc.def.func.a", "def.func.a", "xyz.uvw.func.a"
+    or "uvw.func.a" to refer these two configurable values. You cannot use
+    "func.a" because of the ambiguity. Because of this, you cannot have a config
+    name which is the strict suffix of another config name. For example,
+    "A.Test.arg" and "Test.arg" cannot both be defined.
```

Comment:
What should we do if there is a global function called "a" we want to configure, but there is also another class function "A.a"?

Reply:
I think it would be good to add a section in ALF notes explaining ALF config name scheme on RTD once this PR is merged.

Reply:
You can supply a different name:
```python
@configurable("global_a")
def a():
   ...
```